indexers/txindex/kv/kv.go

```diff
@@ -151,13 +165,14 @@ func (txi *TxIndex) indexEvents(result *abci.TxResult, hash []byte, store store.
 			if attr.GetIndex() {
 				err := store.Set(keyForEvent(compositeTag, attr.Value, result), hash)
 				if err != nil {
-					return err
+					return dmtypes.EventKeys{}, err
 				}
+				eventKeys.Keys = append(eventKeys.Keys, keyForEvent(compositeTag, attr.Value, result))
 			}
 		}
 	}
 
-	return nil
+	return dmtypes.EventKeys{}, nil
```

Comment:
seems like you're always returning her nil vs returning eventKeys

Reply:
fixed